### PR TITLE
FHIR -> HL7 MSH segment

### PR DESCRIPTION
**Pull request by clediggins-usds**
This PR adds the configuration to generate the MSH HL7 segment from a fhir document as well as updates a few missing/incorrect fields in FHIR generation.

Test Steps:
1. Run the fhirdata command on a FHIR file and compare the output to the input hl7 for the MSH segment.

## Changes
- Add extensions for the destination element for receiving facility and application id
- Fixes two FHIR generation items to get correct ids
- adds extensions for receiving application and facility within Destination element
- updates primeCLI to ignore the 5 character delimeter in hl7 files (since we cannot support that at this time)
- updates primCLI to give better hl7 parse failure information

### Testing
- [x] Tested locally?

### CODE REVIEW

**Comment by TurkeyFried** [comment]
There's a little missing on the output HL7: `||||PHLabReport-NoAck^HL7^2.16.840.1.113883.9.11^ISO`

Everything up to that point is perfect.

**Review by TurkeyFried — APPROVED** [review]

File: prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name:
```
Comment (PatriciaPerozoUSDS) [inline prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml:1]: I'm curious if there was a reason destination wasn't mapped in the first pass. Totally not blocking at at all but would want to know if there is any historical knowledge on this.

Comment (PatriciaPerozoUSDS) [reply prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml:1]: Also do we need to update the mapping spreadsheet to reflect that we've added this?

File: prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  type: URL
+  required: true
+  valueOf: MSH.25 | MSH.6
+  expressionType: HL7Spec
+
+extension_1:
+  generateList: true
+  valueOf: extension/Extension
+  expressionType: resource
+  condition: $idStr NOT_NULL && $systemStr EQUALS ISO
```
Comment (carlosfelix2) [inline prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml:16]: FYI: Note we will need to support CLIA soon

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 3]
```diff
@@ -1,6 +1,73 @@
...
+    resource: '%resource.source.extension(%`rsext-source-software-vendor-org`).value.resolve()'
+    value: [ '%resource.extension(%`rsext-assigning-authority-namespace-id`).value' ]
+    hl7Spec: [ MSH-3-1 ]
+
+  - name: sending-application-universal-id
+    value: [ '%resource.source.extension(%`rsext-source-identifier-system`).value.toString().replace(''urn:oid:'','''')' ]
+    hl7Spec: [ MSH-3-2 ]
+
+  - name: sending-application-universal-id-type
+    value: [ '''ISO''' ]
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:13]: You need to check that the ID is an OID.  See example in the SFT segment.

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 4]
```diff
@@ -1,6 +1,73 @@
...
+  - name: sending-facility_namespace-id
+    value: [ '%resource.sender.resolve().name' ]
+    hl7Spec: [ MSH-4-1 ]
+
+  - name: sending-facility-universal-id
+    value: [ '%resource.sender.resolve().identifier.value.toString().replace(''urn:oid:'','''')' ]
+    hl7Spec: [ MSH-4-2 ]
+
+  - name: sending-facility-universal-id-type
+    value: [ '''ISO''' ]
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:25]: You need to check that the ID is an OID. See example in the SFT segment.

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 5]
```diff
@@ -1,6 +1,73 @@
...
+  - name: receiving-application-namespace-id
+    value: [ '%resource.destination.name' ]
+    hl7Spec: [ MSH-5-1 ]
+
+  - name: receiving-application-universal-id
+    value: [ '%resource.destination.extension(%`rsext-receiving-application-identifier`).value.toString().replace(''urn:oid:'','''')' ]
+    hl7Spec: [ MSH-5-2 ]
+
+  - name: receiving-application-universal-id-type
+    value: [ '''ISO''' ]
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:37]: You need to check that the ID is an OID. See example in the SFT segment.

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 6]
```diff
@@ -1,6 +1,73 @@
...
+  - name: receiving-facility-namespace-id
+    value: [ '%resource.destination.endpoint' ]
+    hl7Spec: [ MSH-6-1 ]
+
+  - name: receiving-facility-universal-id
+    value: [ '%resource.destination.extension(%`rsext-receiving-facility-identifier`).value.toString().replace(''urn:oid:'','''')' ]
+    hl7Spec: [ MSH-6-2 ]
+
+  - name: receiving-facility-universal-id-type
+    value: [ '''ISO''' ]
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:49]: You need to check that the ID is an OID. See example in the SFT segment.

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 7]
```diff
@@ -1,6 +1,73 @@
...
+    hl7Spec: [ MSH-10 ]
+
+  - name: processing-id
+    resource: '%resource.meta.extension(%`rsext-source-processing-id`).value'
+    value: [ '%resource.coding.code' ]
+    hl7Spec: [ MSH-11-1 ]
+
+  - name: country-code
+    value: [ '%resource.sender.resolve().address.country' ]
+    hl7Spec: [ MSH-17 ]
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:73]: Missing MSH-21

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 8]
```diff
@@ -1,6 +1,73 @@
...
+
+  - name: receiving-facility-universal-id
+    value: [ '%resource.destination.extension(%`rsext-receiving-facility-identifier`).value.toString().replace(''urn:oid:'','''')' ]
+    hl7Spec: [ MSH-6-2 ]
+
+  - name: receiving-facility-universal-id-type
+    value: [ '''ISO''' ]
+    hl7Spec: [ MSH-6-3 ]
+
+  - name: message-date-time
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:52]: I opened a new bug to try to fix this offset changing on the time.

**Review by carlosfelix2 — APPROVED** [review]

**Pull request merged**